integrationTests/multiShard/smartContract/executingSCCalls_test.go

```diff
@@ -0,0 +1,473 @@
+package smartContract
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test within a network of two shards the following situation
+// 1. Node in first shard deploys a smart contract -> we also make sure that the resulting smart contract address falls within the same shard
+// 2. The same account within the first shard calls the smart contract, we make sure the smart contract is updated and the gas
+//  is substracted from the caller's balance
+func TestProcessSCCallsInMultiShardArchitecture_FirstShard(t *testing.T) {
```

Comment:
function too long, could you split it in 2 ?

Reply:
I extracted the code that could be reused, if still not ok I will try refactor more of it